### ISSUE TITLE
[log-shipper] Logs stuck after upgrade to actual vector

### ISSUE THREAD

**Issue by gyrter** [deckhouse/deckhouse#255]
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Version

v1.25.4

### Expected Behavior

Automatic restart and continue log processing.

### Actual Behavior

Logs stuck. There are 2 possible problems with Reflector:
* kube-master connection problems
* Desync error

### Additional Information

* [Vector issue](https://github.com/vectordotdev/vector/issues/8616)
* [Workaround](https://github.com/vectordotdev/vector/issues/7401#issuecomment-845283750)

### Logs

```shell
Oct 21 13:18:38.107 ERROR source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::instrumenting_watcher: Watch stream failed. error=Recoverable { source: K8sStream { source: Reading { source: hyper::Error(Body, Custom { kind: UnexpectedEof, error: "unexpected EOF during chunk size line" }) } } } internal_log_rate_secs=5
Oct 21 13:18:38.125  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=K8sStream { source: Reading { source: hyper::Error(Body, Custom { kind: UnexpectedEof, error: "unexpected EOF during chunk size line" }) } }
Oct 21 13:18:38.127 ERROR source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::instrumenting_watcher: Internal log [Watch stream failed.] is being rate limited.
Oct 21 13:18:38.127  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=K8sStream { source: Reading { source: hyper::Error(Body, Custom { kind: UnexpectedEof, error: "unexpected EOF during chunk size line" }) } }
Oct 21 13:18:39.157 ERROR source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::instrumenting_watcher: Watch invocation failed. error=Recoverable { source: Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } } } internal_log_rate_secs=5
Oct 21 13:18:39.160  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:39.166 ERROR source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::instrumenting_watcher: Internal log [Watch invocation failed.] is being rate limited.
Oct 21 13:18:39.166  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:40.168  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:40.168  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:41.170  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:41.170  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:42.171  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:42.171  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:43.176  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:43.176  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:44.177 ERROR source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::instrumenting_watcher: Internal log [Watch invocation failed.] has been rate limited 9 times.
Oct 21 13:18:44.177 ERROR source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::instrumenting_watcher: Watch invocation failed. error=Recoverable { source: Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } } } internal_log_rate_secs=5
Oct 21 13:18:44.178  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:44.178 ERROR source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::instrumenting_watcher: Internal log [Watch invocation failed.] is being rate limited.
Oct 21 13:18:44.178  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:45.184  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:45.184  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:46.186  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:46.186  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:47.187  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:47.188  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:48.190  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:48.190  WARN source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::reflector: Http Error in invocation! Your k8s metadata may be stale. Continuing Loop. error=Request { source: CallRequest { source: hyper::Error(Connect, Custom { kind: Other, error: ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }) } }
Oct 21 13:18:54.284 ERROR source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::instrumenting_watcher: Internal log [Watch invocation failed.] has been rate limited 9 times.
Oct 21 13:18:54.284 ERROR source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::internal_events::kubernetes::instrumenting_watcher: Watch invocation failed. error=Other { source: BadStatus { status: 403 } } internal_log_rate_secs=5
Oct 21 13:18:54.284 ERROR source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::kubernetes::reflector: Watcher error. error=BadStatus { status: 403 }
Oct 21 13:18:54.286 ERROR source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::sources::kubernetes_logs: Reflector process exited with an error. error=watch invocation failed
Oct 21 13:18:54.291  INFO source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::sources::kubernetes_logs: Namespace reflector process completed gracefully.
Oct 21 13:18:54.306  INFO source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::sources::kubernetes_logs: Event processing loop completed gracefully.
Oct 21 13:18:54.308  INFO source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::sources::kubernetes_logs: File server completed gracefully.
Oct 21 13:18:54.314  INFO source{component_kind="source" component_id=d8_cluster_whispers-booking-logs component_type=kubernetes_logs component_name=d8_cluster_whispers-booking-logs}: vector::sources::kubernetes_logs: Done.
```

**Comment by gyrter** [comment]
This is vector issue. It may be cause by different problems:
* queue overflowing
* api server failure
* problems with filewatcher.

By Vector team is trying to solve this issues in 0.16 and 0.17 releases. I will upgrade vector Up to 0.17 and check issues. Also I will try to add more graphs to our vector dashboard.

**Comment by gyrter** [comment]
[Related issue in vector repo](https://github.com/vectordotdev/vector/issues/9251)

**Comment by mac2000** [comment]
In our case after deploying Loki and ClusterLogDestination with ClusterLoggingConfig like described in [docs](https://deckhouse.io/en/documentation/v1/modules/460-log-shipper/usage.html#getting-logs-from-all-cluster-pods-and-sending-them-to-loki) nothing happens and Loki does not receive any logs

<details>
<summary>In log shipper controller there are buch of errors like:</summary>
<pre>
{
  "binding": "d8_vector_config",
  "event": "kubernetes",
  "hook": "vector-config-hook.sh",
  "level": "info",
  "msg": "----------------------------------------------------------------------------------------",
  "output": "stdout",
  "queue": "main",
  "task": "HookRun",
  "time": "2021-11-11T08:30:39Z"
}
{
  "binding": "d8_vector_config",
  "event": "kubernetes",
  "hook": "vector-config-hook.sh",
  "level": "info",
  "msg": "x More than one component with name \"d8_cluster_loki\" (source, sink).",
  "output": "stdout",
  "queue": "main",
  "task": "HookRun",
  "time": "2021-11-11T08:30:39Z"
}
{
  "binding": "d8_vector_config",
  "event": "kubernetes",
  "hook": "vector-config-hook.sh",
  "level": "info",
  "msg": "x Cyclic dependency detected in the chain [ d8_cluster_loki -\u003e d8_tf_loki_0 -\u003e d8_tf_loki_1 -\u003e d8_cluster_loki ]",
  "output": "stdout",
  "queue": "main",
  "task": "HookRun",
  "time": "2021-11-11T08:30:39Z"
}
{
  "binding": "d8_vector_config",
  "event": "kubernetes",
  "hook": "vector-config-hook.sh",
  "level": "info",
  "msg": "",
  "output": "stdout",
  "queue": "main",
  "task": "HookRun",
  "time": "2021-11-11T08:30:39Z"
}
{
  "binding": "d8_vector_config",
  "event": "kubernetes",
  "hook": "vector-config-hook.sh",
  "level": "error",
  "msg": "Hook failed. Will retry after delay. Failed count is 1850. Error: vector-config-hook.sh FAILED: exit status 78",
  "queue": "main",
  "task": "HookRun",
  "time": "2021-11-11T08:30:39Z"
}
</pre>
</details>

<details>
<summary>applied resources</summary>
<pre>
apiVersion: deckhouse.io/v1alpha1
kind: ClusterLoggingConfig
metadata:
  name: loki
spec:
  type: KubernetesPods
  destinationRefs:
    - loki
---
apiVersion: deckhouse.io/v1alpha1
kind: ClusterLogDestination
metadata:
  name: loki
spec:
  type: Loki
  loki:
    endpoint: http://loki.loki:3100
</pre>

grafana/loki helm chart values
<pre>
loki:
  enabled: true
  tolerations:
  - key: "dedicated.deckhouse.io"
    operator: "Equal"
    value: "system"
    effect: "NoSchedule"
  persistence:
    storageClassName: local-storage
    enabled: true
    size: 10Gi
</pre>
</details>

> At moment workaround might be to deploy own promtail till this issue will be fixed

**Comment by EvgenySamoylov** [comment]
Resolved in 36e1eb3688ae9e30d69f05a74f6686dc9e4e33ae
